gusto/state.py

```diff
@@ -252,6 +253,27 @@ def __init__(self, mesh, vertical_degree=None, horizontal_degree=1,
         #  Constant to hold current time
         self.t = Constant(0.0)
 
+        # setup logger
+        if output.log_level == "debug":
+            logger.setLevel(logging.DEBUG)
+        elif output.log_level == "info":
+            logger.setLevel(logging.INFO)
+        elif output.log_level == "warning":
+            logger.setLevel(logging.WARNING)
+        else:
+            raise ValueError("log_level must be one of: %s, %s, %s, not %s" % ("info", "debug", "warning", output.log_level))
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter(fmt="%(name)s:%(levelname)s %(message)s"))
+        if self.mesh.comm.rank == 0:
+            logger.addHandler(handler)
+        else:
+            logger.addHandler(logging.NullHandler())
```

Comment:
I would do this setup at top-level in `configuration.py`.  Rather than in `state`.  Oh, because you need access to the communicator.  Why not have a free function that you can call instead of shoving even more code in `__init__` here?